python/hsfs/core/arrow_flight_client.py

```diff
@@ -46,84 +52,201 @@ def close():
+        _logger.debug("Retrieving host URL.")
+        if isinstance(client.get_instance(), client.external.Client):
             external_domain = self._variable_api.get_loadbalancer_external_domain()
             if external_domain == "":
-                raise Exception(
-                    "External client could not locate loadbalancer_external_domain "
-                    "in cluster configuration or variable is empty."
-                )
+                self._missing_variable_name = "loadbalancer_external_domain"
+                return None
             host_url = f"grpc+tls://{external_domain}:5005"
         else:
             service_discovery_domain = self._variable_api.get_service_discovery_domain()
             if service_discovery_domain == "":
-                raise Exception(
-                    "Client could not locate service_discovery_domain "
-                    "in cluster configuration or variable is empty."
-                )
+                self._missing_variable_name = "service_discovery_domain"
+                return None
             host_url = f"grpc+tls://flyingduck.service.{service_discovery_domain}:5005"
+        _logger.debug(
+            f"Connecting to Hopsworks Feature Query Service on host {host_url}"
+        )
+        return host_url
 
+    def _disable_for_session(self, message: Optional[str] = None) -> None:
+        self._disabled_for_session = True
+        if self._missing_variable_name is None and message is None:
+            warnings.warn(
+                "Hospworks Feature Query Service is disabled on cluster. Contact your administrator to enable it."
+            )
+        elif self._missing_variable_name is not None:
+            warnings.warn(
+                "Hopsworks Feature Query Service failed to initialise due to invalid cluster configuration. "
+                f"Missing variable: {self._missing_variable_name}."
+            )
+        else:
+            warnings.warn(
+                f"Client initialisation failed: {message}. Hopsworks Feature Query Service will be disabled for this session."
+                "If you believe this is a transient error, you can call `(hopsworks.)hsfs.reset_offline_query_service_client()`"
+                " to re-enable it."
+            )
+
+    def _initialize_flight_client(self):
+        self._client = client.get_instance()
         (tls_root_certs, cert_chain, private_key) = self._extract_certs(self._client)
         self._connection = pyarrow.flight.FlightClient(
-            location=host_url,
+            location=self.host_url,
             tls_root_certs=tls_root_certs,
             cert_chain=cert_chain,
             private_key=private_key,
             override_hostname="flyingduck.service.consul",
+            generic_options=[
+                (
+                    # https://arrow.apache.org/docs/cpp/flight.html#excessive-traffic
+                    "GRPC_ARG_MIN_RECONNECT_BACKOFF_MS",
```

Comment:
how many times will it retry?

Reply:
This does not trigger retries, however it limits the possibility of retrying faster than that

Reply:
This does not retry, it just prevents retry to be too earger